### DISCUSSION TITLE
Training SoundStream doesn't result proper audio

### DISCUSSION THREAD

**Discussion by amitaie** [lucidrains/audiolm-pytorch#113] [Q&A]
Hey, looking for some help in training SoundStream.

I'm training SoundStream from version 0.15.8 and my results sounds really bad after 20K steps (attached below). furthermore i noticed few  things that i would like to share and hear if that happened to anyone:
1. The EMA results during training is totally noise, while the model output is really bad but sounds like some speech.
2. The loss is very noisy and around 5000-8000, from previous results here i saw that the loss was much higher. attaching tensorboard graphs of the loss, would like to hear if those losses looks alright.
3. I trained the model to more then 100K and the loss exploded (17B++) happened to anyone?

EMA result:
https://user-images.githubusercontent.com/113421133/221514146-271b2c5f-6fb1-4f1d-be40-19637107f691.mp4

Model result:
https://user-images.githubusercontent.com/113421133/221514348-4055a652-521f-4621-bc72-bbf60a0ac637.mp4

![image](https://user-images.githubusercontent.com/113421133/221512814-e2174467-b835-4686-9bae-56c9f101e4c4.png)
![image](https://user-images.githubusercontent.com/113421133/221512884-3b8d6a41-ab7c-4cbf-88f7-7b146f005b98.png)
![image](https://user-images.githubusercontent.com/113421133/221512989-be65c014-f199-4bd3-837d-151589334f58.png)
![image](https://user-images.githubusercontent.com/113421133/221513037-0bb3f516-cc75-4b0a-be9c-4bdf5764c65d.png)
![image](https://user-images.githubusercontent.com/113421133/221513083-bb1c7c88-7bc4-40c2-8baf-31384d51e873.png)
![image](https://user-images.githubusercontent.com/113421133/221513143-043cd1e7-36ee-4d92-8a3f-7a0f30d7fb52.png)

Some technical details on my training  LibriTTS (24000 sample rate, train-clean-360), model strides: (3, 4, 5, 8)
batch_size=4, grad_accum_every=8 and data_max_length_seconds=1.

**Comment by lzl1456** [comment]
about soundstream
i use libi-light training, 50k steps ，data_max_length_seconds = 10s
soundstream = SoundStream(
codebook_size = 1024,
target_sample_hz = 16000,
rq_num_quantizers = 12,
attn_window_size = 128, # local attention receptive field at bottleneck
attn_depth = 2 # 2 local attention transformer blocks - the soundstream folks were not experts with attention, so i took the liberty to add some. encodec went with lstms, but attention should be better
).cuda()

Do you have a better training situation? At present, I train the model to compress and encode and restore it directly. Compared with the original audio, the loss is relatively large. Background noise (sounds like machinery) mixed in
